odoo/addons/base/models/res_company.py

```diff
@@ -17,7 +17,7 @@
 class Company(models.Model):
     _name = "res.company"
     _description = 'Companies'
-    _order = 'sequence, name'
+    _order = 'parent_id desc, sequence, name'
```

Comment:
Shouldn't we use the parent_path order here also?

Reply:
Now that we used parent_path in the accounting migration scripts I think we should either use it here too or just close this PR because parent path is unique so the sequence will be useless in the order

Reply:
`parent_path` is more equivalent to `parent_id, id` which is not what is wanted here AFAIK

Reply:
I'm OK with not changing this here, but then the [other PR](https://github.com/odoo/odoo/pull/182706)(s) must be forward ported to master... and then we need to keep ensuring the companies are fetched with the right ordering for `try_loading` in any future upgrade script. The latter seems OK since it's only there where we currently have a problem. Maybe @william-andre can provide a better outlook.